File: 2023/day05/c/main.c
#include <stdio.h>
#include <string.h>

#include "config.h"
#include "part_a.h"
#include "part_b.h"


int main(int argc, char **argv)
{
    printf("2023 - Day 5\n");

    CONFIG config;

    if(load_config(&config, argc, argv))
    {
        return 1;
    }

    switch(config.run_type)
    {
        case PART_A:
            printf("Running Part A\n");

            printf("Result is: %lu\n", part_a(&config));
            break;
        case PART_A_OPENMP:
            printf("Running Part A OpenCL\n");

            printf("Result is: %lu\n", part_a_openmp(&config));
            break;
        case PART_A_OPENCL:
            printf("Running Part A OpenCL\n");

            printf("Result is: %lu\n", part_a_opencl(&config));
            break;
        case PART_B:
            printf("Running Part B\n");

            printf("Result is: %lu\n", part_b(&config));
            break;
        case PART_B_OPENCL:
            printf("Running Part B OpenCL\n");

            printf("Result is: %lu\n", part_b_opencl(&config));
            break;
        case PART_B_OPENMP:
            printf("Running Part B OpenMP\n");

            printf("Result is: %lu\n", part_b_openmp(&config));
            break;
        default:
            printf("Unsupported run type");
            break;
    }

    return 0;
}

File: 2023/day05/go/src/main.go
package main

import (
	"flag"
	"fmt"
	"strings"

	"example.com/day5/src/parta"
	"example.com/day5/src/partb"
)

func main() {

	// Define flags with name, default value, and usage description
	file_path := flag.String("i", "", "File path to run")
	run_type := flag.String("r", "part_a", "which run to do {part_a, part_b}")

	flag.Parse()

	fmt.Println("Advent of code 2023 - day 05")

	if strings.Compare(*run_type, "part_a") == 0 {

		part_a_result := parta.PartA(*file_path)
		fmt.Println("Part A result: {}", part_a_result)

	} else if strings.Compare(*run_type, "part_b") == 0 {

		part_b_result := partb.PartB(*file_path)
		fmt.Println("Part B result: {}", part_b_result)

	}
}
